anomalib/data/utils/boxes.py

```diff
@@ -3,41 +3,53 @@
 # Copyright (C) 2022 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
-from typing import List, Tuple
+from typing import List, Optional, Tuple
 
 import torch
 from torch import Tensor
 
 from anomalib.utils.cv import connected_components_cpu, connected_components_gpu
 
 
-def masks_to_boxes(masks: Tensor) -> List[Tensor]:
+def masks_to_boxes(masks: Tensor, anomaly_maps: Optional[Tensor] = None) -> Tuple[List[Tensor], List[Tensor]]:
     """Convert a batch of segmentation masks to bounding box coordinates.
 
     Args:
         masks (Tensor): Input tensor of shape (B, 1, H, W), (B, H, W) or (H, W)
+        anomaly_maps (Optional[Tensor], optional): Anomaly maps of shape (B, 1, H, W), (B, H, W) or (H, W) which are
+            used to determine an anomaly score for the converted bounding boxes.
 
     Returns:
         List[Tensor]: A list of length B where each element is a tensor of shape (N, 4) containing the bounding box
             coordinates of the objects in the masks in xyxy format.
+        List[Tensor]: A list of length B where each element is a tensor of length (N) containing an anomaly score for
+            each of the converted boxes.
     """
-    masks = masks.view((-1, 1) + masks.shape[-2:])  # reshape to (B, 1, H, W)
-    masks = masks.float()
+    masks = masks.view((-1, 1) + masks.shape[-2:]).float()  # reshape to (B, 1, H, W)
```

Comment:
What is the original shape here? This is somewhat difficult to read. Could einops be utilized to improve the readability of this?

Reply:
Original shape can be `(B, 1, H, W)`, `(B, H, W)` or `(H, W)`. I've refactored this line to make it a bit more easy to follow. Using einops would be overkill I feel.